### PR TITLE
Mark batch groups as dirty on component change

### PR DESCRIPTION
**Pull request by daredevildave**
When model, sprite or element components are enable/disabled/destroyed.

### CODE REVIEW

File: src/scene/batching.js [hunk 1]
```diff
@@ -1035,20 +1047,26 @@ pc.extend(pc, function () {
...
+            this.scene.layers.getLayerById(layers[i]).removeMeshInstances(batch.model.meshInstances);
+        }
+        batch.model.destroy();
+    };etes.krujgadfjgh
+
     /**
-     * @function
-     * @name pc.BatchManager#destroy
+     * @private
+     * @functionetes.krujgadfjgh
```
Comment (willeastcott) [inline src/scene/batching.js]: Uhm. Uh. Interesting.

**Comment by willeastcott** [comment]
The only concern I have is about performance. Have you determined what the performance impact of this is?

**Comment by daredevildave** [comment]
Performance *is* affected by deleting entities from the batch. However, we need to remove entities from batches if they are disabled/deleted.

**Comment by Glidias** [comment]
What happens if let's say there's a scenerio where i got a bunch of blinking batched entities that constantly switch on and off?  Many batched entitites can come and go, disabled or re-enabled from pool again, etc. Or for every frame they run through some user-culler implementation onPReCull that disables/enables the entities based on visibility to camera? Also, the current batching.js doesn't support custom vertex shaders on material, since they overw-rite the chunks back to defaults when re-cloning the material :(...

Also, i feel 1 pc.Entity per batched item is not really good for super large scaled simulation of small things (like projectiles and such..epic battle simulator, etc..) I'm not sure if it's possible to work directly  by narrowing it down to  plain MeshInstances or just plain GraphNodes per cloned mesh or even just Transforms alone (instead of 1 pc.Entity + 1 pc.Model) . Ie. A Lite version like:   1 Model entity contains 1 Batched group of cloned meshinstances (if there's hierachy) or transforms per mesh (if no  hierachy is needed and all clones are flattened to same level under Model entity).  That way, useless extra infos datas AABBs may even be avoided, even if it means less utility in certain cases.

Back then when i did batching in Flash... i had a ICUller.filterMethod(persitantArrayBufferToFill, srcArrayOfInstances):Int hook to return a value, (zero, minus 1, or positive number), to indicate what/how many mesh cloned hierachies are visible. "-1" means all items from srcArrayOfInstances is visible,  but a postive number indiciates array items filled up to that positive len within persitantArrayBufferToFill. Then, the batch data bone transforms are simply updaed per frame for whatever instances found within the filter and pushed to renderer.

**Pull request merged**